File: server/src/main/java/com/seb44main011/petplaylist/domain/comment/entity/Comment.java
package com.seb44main011.petplaylist.domain.comment.entity;

import com.seb44main011.petplaylist.domain.member.entity.Member;
import com.seb44main011.petplaylist.domain.music.entity.Music;
import com.seb44main011.petplaylist.global.common.BaseTimeEntity;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter @Setter
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class Comment extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long commentId;

    @Column(length = 30, nullable = false)
    private String comment;

    @ManyToOne
    @JoinColumn(name = "MEMBER_ID")
    private Member member;

    @ManyToOne
    @JoinColumn(name = "MUSIC_ID")
    private Music music;
}

File: client/src/types/Comment.ts
import { PageInfo } from "./PageInfo";

export type Comment = {
  commentId: number;
  musicId: number;
  memberId: number;
  name: string;
  comment: string;
  createdAt: string;
  modifiedAt: string;
  profile: string;
};

export type CommentData = {
  data: Comment[];
  pageInfo: PageInfo;
};
